sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilters.scala

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.orc
+
+import org.apache.orc.storage.ql.io.sarg.{PredicateLeaf, SearchArgument, SearchArgumentFactory}
+import org.apache.orc.storage.ql.io.sarg.SearchArgument.Builder
+import org.apache.orc.storage.serde2.io.HiveDecimalWritable
+
+import org.apache.spark.sql.sources.Filter
+import org.apache.spark.sql.types._
+
+/**
+ * Utility functions to convert Spark data source filters to ORC filters.
+ */
+private[orc] object OrcFilters {
```

Comment:
This class looks very similar the one in hive module, what's the difference?

Reply:
Yep. [This](https://github.com/apache/spark/pull/18953/files#diff-6cac9bc2656e3782b0312dceb8c55d47) will show you that explicitly. Mainly, it uses new APIs.
```
-        Some(builder.startAnd().equals(attribute, value).end())
 +        val castedValue = castLiteralValue(value, dataTypeMap(attribute))
 +        Some(builder.startAnd().equals(attribute, getType(attribute), castedValue).end())
```

Reply:
What is the reason we use new APIs? The old APIs are not available? or because of performance?

Reply:
Yes. The API is changed. Old API is not available in new library.